app/models/repo.js

```diff
@@ -141,13 +141,21 @@ const Repo = VcsEntity.extend({
     }, (b) => b.get('repoId') === id);
   }),
 
-  cronJobs: computed('id', function () {
-    let id = this.id;
-    return this.store.filter('cron', {
-      repository_id: id
-    }, (cron) => cron.get('branch.repoId') === id);
+  cronJobs: computed('id', 'fetchCronJobs.lastSuccessful.value', function () {
+    const crons = this.fetchCronJobs.get('lastSuccessful.value');
+    if (!crons) {
+      this.get('fetchCronJobs').perform();
+    }
+    return crons || [];
   }),
 
+  fetchCronJobs: task(function* () {
+    const id = yield this.id;
```

Comment:
Hmm.. Are you sure the `yield` makes sense here? I don't think `this.id` returns promise.

Also it looks like `this.store.filter` should have filter, shouldn't it?

Reply:
Thank you Artur!